mkosi/run.py

```diff
@@ -374,11 +390,25 @@ def preexec() -> None:
             make_foreground_process(new_process_group=False)
 
 
-def find_binary(*names: PathString, root: Optional[Path] = None) -> Optional[Path]:
+def find_binary(*names: PathString, root: Path = Path("/")) -> Optional[Path]:
+    path = (
+        ":".join(os.fspath(p)
+        for p in [root / "usr/bin", root / "usr/sbin"])
+        if root != Path("/")
+        else os.environ["PATH"]
+    )
+
     for name in names:
-        path = ":".join(os.fspath(p) for p in [root / "usr/bin", root / "usr/sbin"]) if root else os.environ["PATH"]
+        if Path(name).is_absolute():
+            name = root / Path(name).relative_to("/")
+        elif "/" in str(name):
+            name = root / name
+
         if (binary := shutil.which(name, path=path)):
-            return Path("/") / Path(binary).relative_to(root or "/")
+            if root != Path("/") and not Path(binary).is_relative_to(root):
```

Comment:
If the first clause is true, then path is root/usr/bin:root/usr/sbin and which will only look for name there. I don't think they can then not be relative to root or am I missing some edge case in which?

Reply:
@behrmann Absolute symlinks ;)

Reply:
How I always forget about the cosmic horror unleashed upon the world by symlinks.